db/migrate/20190729170013_split_storages_per_ems.rb

```diff
@@ -0,0 +1,62 @@
+class SplitStoragesPerEms < ActiveRecord::Migration[5.0]
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class HostStorage < ActiveRecord::Base
+    belongs_to :host,    :class_name => "::SplitStoragesPerEms::Host"
+    belongs_to :storage, :class_name => "::SplitStoragesPerEms::Storage"
+  end
+
+  class Storage < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+  end
+
+  def up
+    Storage.in_my_region.find_each do |storage|
+      # First we need to group HostStorage records by their ems_id and ems_ref since these will
+      # form the new Storage records. Once the Storage records are created we can link them
+      # to the hosts through these HostStorage records.
+      host_storages_by_ems_id_and_ems_ref = HostStorage.includes(:host).where(:storage_id => storage.id).group_by do |hs|
+        ems_id            = hs.host&.ems_id
+        datastore_ems_ref = hs.ems_ref || storage.ems_ref
+
+        next if ems_id.nil?
```

Comment:
*possibly* this `next` should happen in a `map` and not a `group_by`?

Reply:
Yeah agreed